pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +275,44 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	if spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
+		for _, option := range mount.Options {
+			if isParentDir(filePath, mount.Destination) {
+				if option == "ro" {
+					return "", "", fmt.Errorf("mount point %s is marked read-only", filePath)
```

Comment:
This seems to correctly checked only when ro is the first option? Is this an intended behaviour?

Reply:
Good catch! Updated.